test/access_control_test.go

```diff
@@ -1286,6 +1455,140 @@ var _ = Describe("AccessControl", func() {
+							Privileges: []string{
+								"read-write.test",
+								"read-write-udf.test.users",
+							},
+						},
+						{
+							Name: "roleToDrop",
+							Privileges: []string{
+								"read-write.test",
+								"read-write-udf.test.users",
+							},
+							Whitelist: []string{
+								"8.8.0.0/16",
+							},
+							ReadQuota:  1,
+							WriteQuota: 1,
+						},
+					},
+					Users: []asdbv1alpha1.AerospikeUserSpec{
+						{
+							Name:       "admin",
+							SecretName: authSecretName,
+							Roles: []string{
+								"sys-admin",
+								"user-admin",
+							},
+						},
+
+						{
+							Name:       "profileUser",
+							SecretName: authSecretName,
+							Roles: []string{
+								"profiler",
+								"sys-admin",
+							},
+						},
+
+						{
+							Name:       "userToDrop",
+							SecretName: authSecretName,
+							Roles: []string{
+								"profiler",
+							},
+						},
+					},
+				}
+
+				aerospikeConfigSpec, err = NewAerospikeConfSpec(ver)
+				if err != nil {
+					Fail(fmt.Sprintf("Invalid Aerospike Config Spec: %v", err))
+				}
+				aerospikeConfigSpec.setEnableSecurity(true)
+				aerospikeConfigSpec.setEnableQuotas(true)
+
+				aeroCluster = getAerospikeClusterSpecWithAccessControl(
+					clusterNamespacedName, &accessControl, aerospikeConfigSpec, ctx)
+				err = testAccessControlReconcile(aeroCluster, ctx)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("QuotaParamsSpecifiedButFlagIsOff")
+
+				accessControl = asdbv1alpha1.AerospikeAccessControlSpec{
+					Roles: []asdbv1alpha1.AerospikeRoleSpec{
+						{
+							Name: "profiler",
+							Privileges: []string{
+								"read-write.test",
+								"read-write-udf.test.users",
+							},
+						},
+						{
+							Name: "roleToDrop",
+							Privileges: []string{
+								"read-write.test",
+								"read-write-udf.test.users",
+							},
+							Whitelist: []string{
+								"8.8.0.0/16",
+							},
+							ReadQuota:  1,
+							WriteQuota: 1,
+						},
+					},
+					Users: []asdbv1alpha1.AerospikeUserSpec{
+						{
+							Name:       "admin",
+							SecretName: authSecretName,
+							Roles: []string{
+								"sys-admin",
+								"user-admin",
+							},
+						},
+
+						{
+							Name:       "profileUser",
+							SecretName: authSecretName,
+							Roles: []string{
+								"profiler",
+								"sys-admin",
+							},
+						},
+
+						{
+							Name:       "userToDrop",
+							SecretName: authSecretName,
+							Roles: []string{
+								"profiler",
+							},
+						},
+					},
+				}
+
+				aerospikeConfigSpec, err = NewAerospikeConfSpec(ver)
+				if err != nil {
+					Fail(fmt.Sprintf("Invalid Aerospike Config Spec: %v", err))
+				}
+				aerospikeConfigSpec.setEnableSecurity(true)
+				aerospikeConfigSpec.setEnableQuotas(false)
+
+				aeroCluster = getAerospikeClusterSpecWithAccessControl(
+					clusterNamespacedName, &accessControl, aerospikeConfigSpec, ctx)
+				err = testAccessControlReconcile(aeroCluster, ctx)
```

Comment:
In validateRoles I think you need to check that the read and write quotas are matching betweenspec and what is reported by the client.

Reply:
Done